src/wayland/pointer_constraints.rs

```diff
@@ -171,14 +183,18 @@ impl PointerConstraint {
         }
     }
 
-    fn commit(&mut self) {
+    fn commit(&mut self) -> Option<Point<f64, Logical>> {
         match self {
             Self::Confined(confined) => {
                 confined.region.clone_from(&confined.pending_region);
+                None
             }
             Self::Locked(locked) => {
                 locked.region.clone_from(&locked.pending_region);
-                locked.cursor_position_hint = locked.pending_cursor_position_hint;
+                locked.pending_cursor_position_hint.take().map(|hint| {
+                    locked.cursor_position_hint = Some(hint);
+                    hint
+                })
```

Comment:
Seems a little bit overcomplicated for what it does, isn't is essentially this:
```suggestion
                locked.cursor_position_hint = locked.pending_cursor_position_hint.take();
                locked.cursor_position_hint
```